membership-attribute-service/test/parsers/SalesforceTest.scala

```diff
@@ -110,5 +191,25 @@ class SalesforceTest extends Specification {
 
       Salesforce.parseOutboundMessage(payload, orgId) shouldEqual OrgIdMatchingError("wrong org").left
     }
+
+    "returns an error if there are no Salesforce Objects" in {
+      val payload =
+        <soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+          <soapenv:Body>
+            <notifications xmlns="http://soap.sforce.com/2005/09/outbound">
+              <OrganizationId>{orgId}</OrganizationId>
+              <ActionId>action_id</ActionId>
+              <SessionId xsi:nil="true"/>
+              <EnterpriseUrl>https://cs17.salesforce.com/services/Soap/c/34.0/enterprise_id</EnterpriseUrl>
+              <PartnerUrl>https://cs17.salesforce.com/services/Soap/u/34.0/enterprise_id</PartnerUrl>
+              <Notification>
+                <Id>notification_id</Id>
+              </Notification>
+            </notifications>
+          </soapenv:Body>
+        </soapenv:Envelope>
+
+      Salesforce.parseOutboundMessage(payload, orgId) shouldEqual ParsingError(s"No Salesforce Objects were found when parsing the message. \n $payload").left
```

Comment:
Yes, that's possible. We don't respond to Salesforce until after we've actually actioned all of the updates (in SalesforceHookController.scala) and as there doesn't seem to be any way of responding with granularity about what did/didn't succeed, we just return an API error if there are any failures during the update sequence.